### PR TITLE
Classification head remove

### PR DESCRIPTION
**Pull request by BrandonTatani**
I have removed the classification head from the network. However, the entire structure of the collimator angle still remains, which in my opinion is a bit useless since we carry that information with us a priori, so in my opinion it can be modified.

Regarding the network output in the 90 degree collimator, for the two fields on the legs 7 degrees of freedom are used while in the case of fields at 5/355, given the symmetry between the two fields, there would only be 5 degrees of freedom, of which one removable in my opinion under the condition of minimum overlap of the two fields.

### CODE REVIEW

File: src/models/cnn.py [hunk 1]
```diff
@@ -63,7 +58,5 @@ def forward(self, x: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
                 and classification heads
```
Comment (nlambriICH) [inline src/models/cnn.py:58]: Remove "classification" from the docstring.

File: src/utils/visualization_cnn.py
Comment (nlambriICH) [inline src/utils/visualization_cnn.py:1]: When joining paths, use `os.path.join` instead of string concatenation.

File: src/data/processing.py [hunk 2]
```diff
@@ -3,6 +3,14 @@
 from imgaug.augmentables import Keypoint, KeypointsOnImage
 from src.utils.field_geometry_transf import get_zero_row_idx
 import os
+from src.config.constants import CLASSIFICATION
+
+
+def directory() -> str:
```
Comment (nlambriICH) [inline src/data/processing.py:9]: Move this funciton in utils. Make a new python file if necessary. Return the absolute path.

File: src/modules/lightning_cnn.py [hunk 3]
```diff
@@ -164,21 +142,14 @@ def test_step(  # pylint: disable=arguments-differ
...
-            y_reg_hat = self.cnn(x)
-            y_cls_hat = torch.zeros(1)
-            test_mse_loss = self.weighted_mse_loss(y_reg_hat, y_reg)
-            metrics = {
-                "test_mse_loss": test_mse_loss,
-            }
-            y_train_reg_hat = self.cnn(x_train)
-            y_train_cls_hat = torch.zeros(1)
+        y_reg_hat = self.cnn(x)
+        y_cls_hat = torch.zeros(1)
```
Comment (nlambriICH) [inline src/modules/lightning_cnn.py:146]: This is 0 for collimator=0 deg, and 1 for collimator=5/355 deg. You can set `y_cls_hat` and `y_train_cls_hat` to either 0 or 1 based on `COLL_5_355`.

Then, these are passed to `plot_img`. Change this [line](https://github.com/nlambriICH/tmi-isocenter/blob/main/src/utils/visualization_cnn.py#L501) by removing the sigmoid function (we don't need it as we already have either 0 or 1).

File: src/modules/lightning_cnn.py [hunk 4]
```diff
@@ -164,21 +142,14 @@ def test_step(  # pylint: disable=arguments-differ
...
+        test_mse_loss = self.weighted_mse_loss(y_reg_hat, y_reg)
+        metrics = {
+            "test_mse_loss": test_mse_loss,
+        }
+        y_train_reg_hat = self.cnn(x_train)
+        y_train_cls_hat = torch.zeros(1)
```
Comment (nlambriICH) [inline src/modules/lightning_cnn.py:152]: See previous comment.

**Review by nlambriICH — CHANGES_REQUESTED** [review]
After the changes you can do a dry run to check for potential bugs.

File: src/data/dataset.py [hunk 5]
```diff
@@ -67,21 +81,18 @@ def train_val_test_split(
             - 10% for validation
 
         Args:
-            test_set (str): The strategy to split the test data based on the  class labels (collimator angle).
+            test_set (str)
```
Comment (nlambriICH) [inline src/data/dataset.py:84]: Remove Args from docstring

File: src/modules/lightning_cnn.py [hunk 6]
```diff
@@ -1,13 +1,14 @@
 """Lightning module for CNN training"""
+import lightning.pytorch as pl
 import torch
 import torch.nn.functional as F
-import lightning.pytorch as pl
-from torch.optim.lr_scheduler import ReduceLROnPlateau
 from torch.optim import Adam
+from torch.optim.lr_scheduler import ReduceLROnPlateau
 from torchmetrics.classification import BinaryAccuracy
+
```
Comment (nlambriICH) [inline src/modules/lightning_cnn.py:8]: Remove unused imports

File: src/train_cnn.py [hunk 7]
```diff
@@ -69,29 +64,27 @@
         TensorDataset(
             masks_train,
             y_reg_train,
-            y_cls_train,
         ),
-        num_workers=16,
+        num_workers=4,
```
Comment (nlambriICH) [inline src/train_cnn.py:68]: Create a global config in config.py which is read from config.json

**Review by nlambriICH — CHANGES_REQUESTED** [review]

**Review by nlambriICH — APPROVED** [review]
